torch/_dynamo/output_graph.py

```diff
@@ -266,6 +269,21 @@ def __init__(
         # create unique node names
         self.input_name_to_proxy: OrderedDict[str, fx.Proxy] = collections.OrderedDict()
 
+        self.proxies = set({})
+
+    def new_subgraph(self):
+        result = OutputGraph(
+            self.root_globals,
+            self.code_options,
+            self.compiler_fn,
+            self.root_tx,
+            self.export,
+            self.export_constraints,
+            self.frame_state,
+            self)
+        result.side_effects = self.side_effects
+        return result
```

Comment:
Instead of having nested OutputGraph, I'm thinking of splitting OutputGraph into {OutputGraph and NestableGraphTracer}.

Nested OutputGraph doesn't seem like the correct abstraction because there are methods on it (like compile_subgraph, add_symbol_bindings) that should always apply to root OutputGraph and never to the body of a higher order operator.

Thoughts?

Reply:
Up to you on abstraction - If you end up using most of OutputGraph, nesting is still cleaner, and you'll just have to have compile_subgraph/etc raise if there is a parent set. If the raise pattern is the norm (as in, you end up not using most of OutputGraph), I would make another class as you suggested.